packages/itwin/tree-widget/src/tree-widget-react/components/trees/categories-tree/CategoriesTreeButtons.tsx

```diff
@@ -121,7 +123,8 @@ export function InvertAllButton(props: CategoriesTreeHeaderButtonProps) {
 
 const EMPTY_CATEGORIES_ARRAY: CategoryInfo[] = [];
 
-export function useCategories(viewport: Viewport) {
-  const categoriesPromise = useMemo(async () => loadCategoriesFromViewport(viewport), [viewport]);
+function useCategories(viewport: Viewport) {
+  const componentId: GuidString = useMemo(() => Guid.createValue(), []);
```

Comment:
I think non-exported hooks / components should take id as an argument and not create their own.